sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -114,7 +114,16 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
     }
 
     (file: PartitionedFile) => {
-      val parser = new JacksonParser(actualSchema, parsedOptions)
+      // SPARK-21610: when the `requiredSchema` only contains `_corrupt_record`,
```

Comment:
> The column _corrupted_record is different when the selected columns are different

If `_corrupted_record` is designed to have different values for different selected columns, it may makes sense to set `_corrupted_record` to null if no columns are selected.